### DISCUSSION TITLE
Combobox responsiveness and mobile advice

### DISCUSSION THREAD

**Discussion by amrita-sakhrani** [twilio-labs/paste#2822] [General] [answered]
Hi. The growth team is working on a redesign of the sign-up page in order to migrate this page to Paste and move towards a future visual alignment with the login page. As part of this larger effort, the localization team is also looking to introduce a language selector on the sign-up, with more languages to be added next year. 

I wanted to ask the for any recommendations or guidelines for using the combobox on the new sign-up screen, and how this combobox component would behave on a mobile screen? Any key considerations to keep in mind and if this is the best approach for a language selector on the sign-up? Thank you 

A couple of relevant Figma frames can be found here: 
https://www.figma.com/file/Bmr7ltHScX620smbk0DBpc/Signup-%2F-try-twilio-migration-to-Paste-(2022)?node-id=1040%3A265400 
https://www.figma.com/file/Bmr7ltHScX620smbk0DBpc/Signup-%2F-try-twilio-migration-to-Paste-(2022)?node-id=1016%3A211113

**Comment by SiTaggart** [comment, marked as answer]
Hi @amrita-sakhrani 

For this use casee the component I'm actually going to suggest you use in the [Select](https://paste.twilio.design/components/select/#whats-the-difference-between-select-and-combobox).

This is for a couple of reasons:

- The collection of data seems relatively small, we're only supporting a few languages
- The native mobile experience is tailored to touch screen devices automatically for you

For those two reasons, this would seem like the most suitable Component to use. If the number of languages starts to grow, say larger than 20, then maybe we can discuss how you would use combobox, but for now I think Select is what you're after.

Thanks!

**Reply by amrita-sakhrani** [reply]
Hi @SiTaggart Thank you for your suggestion on this. I will go ahead to use the select for the sign-up language selector that will be upcoming as we introduce languages in the platform. Please see here for reference to see if it is alright: 
https://www.figma.com/file/Bmr7ltHScX620smbk0DBpc/Signup-%2F-try-twilio-migration-to-Paste-(2022)?node-id=1151%3A273514

**Reply by SiTaggart** [reply]
Yeah, that looks perfect!

**Reply by rosemaryluo** [reply]
Back again with another combobox -> select question!
We are rolling with this grouped design for a product selection dropdown on desktop.
We'd like to keep the groupings in the mobile experience, but select doesn't offer grouping. Is this something our team could modify/build on or is this a limitation of the native experience?
<img width="522" alt="image" src="https://user-images.githubusercontent.com/87049408/205162630-1ec2f9d1-d8c6-450b-9ffa-2e837a9217fc.png">

**Reply by SiTaggart** [reply]
Hi

Select does offer groupings https://paste.twilio.design/components/select/#select-with-option-groups so you're covered. 👍🏼

**Reply by rosemaryluo** [reply]
@SiTaggart lifesaving. Don't know how I missed that in my readthrough. Thanks!

**Comment by amrita-sakhrani** [comment]
Thank you so much. We have another follow-up question. @SiTaggart We are also looking to introduce a language selector inside the Console, which will eventually and ideally sit inside the user settings page. However, the user settings page is currently not yet been pastified, so we have a temporary interim modal solution until this selector can be migrated into the new future user settings page. 

For a first version, we will only have two languages. However, we are looking to introduce a badge at some point in the future, to indicate a new or beta language. I am working with Giammarco from the Localization team on this. We are having an issue where the combobox selector does not overflow from the modal. If you click into the language preferences within modal at the bottom here, hopefully you can understand the issue https://b5phd4.csb.app/ 

We want to ask for your advice and guidance on this? If we were to introduce a badge next to the language in the future as localization scales and more languages are to be added.

**Comment by SiTaggart** [comment]
Hi @amrita-sakhrani 

That's a known issue and is in our backlog to address.

In the meantime, as you only have a small modal and a couple of options, a really quick and safe option would be to just add a min-height to the modal body via a `Box`, so that when the combobox is opened, the options are still in view.

It's a workaround until we get to the Combobox options work.

**Comment by rosemaryluo** [comment]
Oh, don’t know how I missed that! Thanks!

On Thu, Dec 1, 2022 at 2:09 PM Simon Taggart ***@***.***>
wrote:

> Hi
>
> Select does offer groupings
> https://paste.twilio.design/components/select/#select-with-option-groups
> <https://urldefense.com/v3/__https://paste.twilio.design/components/select/*select-with-option-groups__;Iw!!NCc8flgU!bTuunS0ArgNwt3YleXqc7Y7c_13RZ-WCK-3Hu0C0xjBk_wvgg9i9ZAmD8_NC_0-V56BW0-iA9OPgcxfHE4JY$>
> so you're covered. 👍🏼
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/twilio-labs/paste/discussions/2822*discussioncomment-4287401__;Iw!!NCc8flgU!bTuunS0ArgNwt3YleXqc7Y7c_13RZ-WCK-3Hu0C0xjBk_wvgg9i9ZAmD8_NC_0-V56BW0-iA9OPgc_9Hvbjt$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AUYEJQAVCGN3NHZGOLXASXDWLEOXDANCNFSM6AAAAAAR4OZ5BQ__;!!NCc8flgU!bTuunS0ArgNwt3YleXqc7Y7c_13RZ-WCK-3Hu0C0xjBk_wvgg9i9ZAmD8_NC_0-V56BW0-iA9OPgc8SrmjoS$>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 

Rosemary Luo (she/her)
Staff Product Designer, Platform Product Design
[image: Twilio] <https://www.twilio.com/?utm_source=email_signature>
EMAIL ***@***.***
<https://surveys.twilio.com/jfe/form/SV_bOcOhWU9iEVH4zj?&source=esig> Join
Twilio Research to access beta products, work with the Twilio team, and be
a part of paid studies.
<https://surveys.twilio.com/jfe/form/SV_bOcOhWU9iEVH4zj?&source=esig>
